src/Serialization/containers.jl

```diff
@@ -114,12 +106,12 @@ end
 function load_object(s::DeserializerState, T::Type{<: Vector{params}}) where params
   load_node(s) do v
     if serialize_with_id(params)
-      loaded_v::Vector{params} = load_array_node(s) do _
+      loaded_v = load_array_node(s) do _
         load_ref(s)
-      end
+      end::Vector{params}
     else
       isempty(s.obj) && return params[]
-      loaded_v = load_array_node(s) do obj
+      loaded_v = load_array_node(s) do _
         load_object(s, params)
       end
```

Comment:
Shouldn't this also have a type assertion?
```suggestion
      end::Vector{params}
```

See also [my older comments](https://github.com/oscar-system/Oscar.jl/issues/4558#issuecomment-3270049323) on type stability in the serialization code (and the comments after that)

Reply:
this seems good in this case, yes.

I think things get trickier when the load has paramters being passed to it.  but maybe not